File: spring-httpclient/src/main/java/com/example/httpclient/config/RestServerConfig.java
package com.example.httpclient.config;

import java.nio.file.Files;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.example.httpclient.model.RestServerConfigDto;
import com.example.httpclient.util.PathUtil;
import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Configuration
@Slf4j
@RequiredArgsConstructor
public class RestServerConfig {

    private final YAMLMapper yamlMapper;

    /*
     * read rest server information to request from "rest-server.yaml"
     */
    @Bean
    public RestServerConfigDto RestServerInfo() {
        var path = PathUtil.processWorkingDirectory().resolve(
            "rest-server.yaml"
        ).normalize();

        log.info(path.toString());

        if (!(Files.exists(path) && Files.isRegularFile(path))) {
            log.error("can NOT find configuration file");
        }

        RestServerConfigDto yaml = new RestServerConfigDto();
        try {
            yaml = yamlMapper.readValue(path.toFile(), RestServerConfigDto.class);
        } catch(Exception e) {
            log.error(e.getMessage(), e);
        }

        return yaml;
    }
}

File: spring-httpclient-kotlin/src/main/kotlin/com/example/httpclient/config/RestServerConfig.kt
package com.example.httpclient.config

import com.example.httpclient.model.RestServerConfigDto
import com.example.httpclient.util.PathUtil
import tools.jackson.dataformat.yaml.YAMLMapper
import io.github.oshai.kotlinlogging.KotlinLogging
import java.nio.file.Files
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class RestServerConfig(
    private val yamlMapper: YAMLMapper,
) {
    private val log = KotlinLogging.logger {}

    @Bean
    fun restServerInfo(): RestServerConfigDto {
        val path = PathUtil.processWorkingDirectory()
            .resolve("rest-server.yaml")
            .normalize()

        log.info { path.toString() }

        if (!(Files.exists(path) && Files.isRegularFile(path))) {
            log.error { "can NOT find configuration file" }
        }

        return try {
            yamlMapper.readValue(path.toFile(), RestServerConfigDto::class.java)
        } catch (e: Exception) {
            log.error(e) { e.message ?: "failed to read rest-server.yaml" }
            RestServerConfigDto()
        }
    }
}
